mathics/builtin/numbers/algebra.py

```diff
@@ -405,44 +403,59 @@ class Simplify(Builtin):
-        "Simplify[rule_Rule]": "Simplify /@ rule",
-        "Simplify[list_List, assum_]": "Simplify[#1, assum]& /@ list",
-        "Simplify[rule_Rule, assum_]": "Simplify[#1, assum]& /@ rule",
-        "Simplify[0^a_, assum_]": "ConditionalExpression[0,Simplify[a>0, assum]]",
-        "Simplify[b_^a_, assum_]": "ConditionalExpression[b^a,Simplify[Or[a>0, b!=0],assum]]",
+        "Simplify[list_List, opts:OptionsPattern[]]": "Simplify[#1,opts]& /@ list",
+        "Simplify[rule_Rule, opts:OptionsPattern[]]": "Simplify[#1,opts]& /@ rule",
+    }
+    options = {
+        "Assumptions": "$Assumptions",
     }
 
-    def apply_assuming(self, expr, assumptions, evaluation):
-        "%(name)s[expr_, assumptions_]"
-        assumptions = assumptions.evaluate(evaluation)
-        return dynamic_scoping(
-            lambda ev: self.apply(expr, ev),
-            {"System`$Assumptions": assumptions},
-            evaluation,
-        )
-
-    def apply(self, expr, evaluation):
-        "%(name)s[expr_]"
+    def apply_power_of_zero(self, b, evaluation, options={}):
+        "%(name)s[0^b_, OptionsPattern[]]"
+        if self.apply(Expression("Less", 0, b), evaluation, options) is SymbolTrue:
+            return Integer0
+        if self.apply(Expression("Less", b, 0), evaluation, options) is SymbolTrue:
+            return Symbol("ComplexInfinity")
+        if self.apply(Expression("Equal", b, 0), evaluation, options) is SymbolTrue:
+            return Symbol("Indeterminate")
+        return Expression("Power", Integer0, b)
+
+    def apply(self, expr, evaluation, options={}):
+        "%(name)s[expr_, OptionsPattern[]]"
         # Check first if we are dealing with a logic expression...
         if expr in (SymbolTrue, SymbolFalse, SymbolList):
             return expr
+
+        # Check if we need to consider special assumptions.
+        if options:
+            assumptions_list = options.get("System`Assumptions", None)
+            if assumptions_list is not Symbol("$Assumptions"):
+                assumptions = assumptions_list.evaluate(evaluation)
+                return dynamic_scoping(
+                    lambda ev: self.apply(expr, ev, {}),
+                    {"System`$Assumptions": assumptions},
+                    evaluation,
+                )
+        # Now, assumptions are contained in Symbol("$Assumptions")
         expr = evaluate_predicate(expr, evaluation)
         if expr.is_atom():
             return expr
         # else, use sympy:
-        elements = [self.apply(element, evaluation) for element in expr._elements]
-        head = self.apply(expr.get_head(), evaluation)
+        # Here we want to pass through the full evaluation process to use all the defined rules...
+        elements = [
+            Expression("Simplify", element).evaluate(evaluation)
+            for element in expr._elements
+        ]
+        head = Expression("Simplify", expr.get_head()).evaluate(evaluation)
```

Comment:
SymbolSimplify

Reply:
Better is ``self.get_name()`` to handle also ``FullSymplify``